### PR TITLE
Fix and enable x86 build for Android

### LINKED ISSUE #11744: Android apk x86 build
Hello.
Would be possible to have a x86 apk build for android, so it can be used on x86 tablets. Thanks you!

### PR DESCRIPTION
**Pull request by savilli**
See minetest/minetest_android_deps#2. Should be merged after that PR.

Fixes #11744

### CODE REVIEW

**Comment by rollerozxa** [comment]
Should the x86 APK be split as well with a different version code like the armeabi-v7a and arm64 versions are currently?

**Comment by sfan5** [comment]
A word of caution: it will be impossible to upload the x86 build to the playstore without also providing a x86_64 build, since the [requirement](https://developer.android.com/google/play/requirements/64-bit) also applies here.

File: android/native/jni/Android.mk [hunk 1]
```diff
@@ -281,7 +291,8 @@ ifndef USE_BUILTIN_LUA
 endif
 LOCAL_STATIC_LIBRARIES += android_native_app_glue $(PROFILER_LIBS)
 
-LOCAL_LDLIBS := -lEGL -lGLESv1_CM -lGLESv2 -landroid -lOpenSLES
+# Irrlicht uses zlib
```
Comment (sfan5) [inline android/native/jni/Android.mk:294]: (we also use it ourselves)
```suggestion
```

**Comment by savilli** [comment]
> A word of caution: it will be impossible to upload the x86 build to the playstore without also providing a x86_64 build, since the [requirement](https://developer.android.com/google/play/requirements/64-bit) also applies here.

A good point. Enabled x86_64 build as well.

**Comment by sfan5** [comment]
Well I think we're fine with just x86 and people who have one of these rare devices can download the right build from Github but :shrug:

**Comment by savilli** [comment]
I think it won't be excessive.

**Comment by rubenwardy** [comment]
> Should the x86 APK be split as well with a different version code like the armeabi-v7a and arm64 versions are currently?

It would be nice to be able to have one version code for all the variants. I upload the app to play using an app bundle already, so it might be possible

**Review by sfan5 — APPROVED** [review]

**Review by rubenwardy — DISMISSED** [review]

**Comment by rubenwardy** [comment]
You forgot to enable x86 and x86_64 in android/native/build.gradle

**Comment by savilli** [comment]
> You forgot to enable x86 and x86_64 in android/native/build.gradle

Where?

**Comment by rubenwardy** [comment]
In a x86_64 emulator: 

image

**Comment by savilli** [comment]
> In a x86_64 Android 13 emulator. Could just be an emulator graphics acceleration issue, I don't have a x86 device

Looks like it's out of GPU memory, try to increase it.

**Comment by rubenwardy** [comment]
> Looks like it's out of GPU memory, try to increase it.

Any idea how to do that? There's only options for RAM and storage

image

**Comment by rubenwardy** [comment]
If someone can test this on an actual x86 device and it works them I'm happy

**Comment by savilli** [comment]
> > Looks like it's out of GPU memory, try to increase it.
> 
> Any idea how to do that? There's only options for RAM and storage

No idea, I've managed to run it without graphic issues with an image from https://www.android-x86.org/ and a long qemu command.

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**